public/app/plugins/datasource/prometheus/func_editor.js

```diff
@@ -0,0 +1,166 @@
+define([
+  'angular',
+  'lodash',
+  'jquery',
+],
+function (angular, _, $) {
+  'use strict';
+
+  angular
+    .module('grafana.directives')
+    .directive('prometheusFuncEditor', function($compile) {
+
+      var exprTemplate = '<input type="text" class="input-xxlarge tight-form-input"' +
+                         'spellcheck="false" placeholder="query expression" data-min-length=0 data-items=100></input>';
+
+      return {
+        restrict: 'A',
+        link: function postLink($scope, elem) {
+          var ctrl = $scope.ctrl;
+
+          function inputBlur() {
+            /* jshint validthis:true */
+            var $input = $(this);
+            ctrl.target.expr = $input.val();
+            ctrl.refreshMetricData();
+          }
+
+          function inputKeyPress(e) {
+            /* jshint validthis:true */
+            if (e.which === 13) {
```

Comment:
I don't know well about event code, but Grafana already have such magic number, https://github.com/grafana/grafana/blob/75555c470b695fb196fba7256a6c7b30ecb340a4/public/app/core/directives/metric_segment.js#L131.

And this code is originally written by @juliusv , I just porting the code to Grafana.